app/controllers/v1/pseudonym.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const consoleLogger = require('../../utils/consoleLogger');
+const db = require('../../services/db');
+const env = require('../../../env');
+const crypto = require('../../utils/crypto');
+
+function sendResponse(req, res, status, json) {
+	var isJsonP = req.query.callback ? true : false;
+
+	status = status || 200;
+
+	res.status(isJsonP ? 200 : status).jsonp(json);
+}
+
+function validateApiKey(req, res, callback) {
+	const apiKey = req.headers['x-api-key'] || req.query.apiKey;
+	if (apiKey) {
+		if (apiKey !== env.apiKeyForRestrictedEndpoints) {
+			sendResponse(req, res, 401, {
+				error: 'The API key is invalid.'
+			});
+			return false;
+		} else {
+			return true
+		}
+	} else {
+		sendResponse(req, res, 400, {
+			error: 'The API key is missing.'
+		});
+		return false;
+	}
+}
+
+exports.available = function (req, res) {
+	var pseudonym = req.params.pseudonym;
+
+	if (validateApiKey(req, res)) {
+		if (!pseudonym) {
+			return res.status(400).json({
+				success: false,
+				error: 'Pseudonym is missing'
+			});
+		}
+
+		db.getConnection(env.mongo.uri, function (errConn, connection) {
+			if (errConn) {
+				consoleLogger.warn(pseudonym, 'action failed', errConn);
```

Comment:
I'm not completely sure what the logger is warning here so I added a generic `action failed`, I'm open to suggestions!